atlasdb-config/src/main/java/com/palantir/atlasdb/factory/TransactionManagers.java

```diff
@@ -818,13 +824,15 @@ private static LockAndTimestampServices createRawServicesFromTimeLock(
     private static LockAndTimestampServices getLockAndTimestampServices(
             MetricsManager metricsManager,
             Supplier<ServerListConfig> timelockServerListConfig,
-            String userAgent) {
+            String userAgent,
+            List<AutoCloseable> closeables) {
         ServiceCreator creator = ServiceCreator.withPayloadLimiter(metricsManager, userAgent, timelockServerListConfig);
         LockService lockService = creator.createService(LockService.class);
         TimelockRpcClient timelockClient = creator.createService(TimelockRpcClient.class);
-        TimelockService timelockService = RemoteTimelockServiceAdapter.create(timelockClient);
+        RemoteTimelockServiceAdapter timelockService = RemoteTimelockServiceAdapter.create(timelockClient);
         TimestampManagementService timestampManagementService = creator.createService(TimestampManagementService.class);
 
+        closeables.add(timelockService::close);
```

Comment:
nit: I think you can just do `RemoteTimelockServiceAdapter timelockServiceAdapter = initializeCloseable(() -> RemoteTimelockServiceAdapter.create(timelockClient), closeables);`

Also I guess I don't really like naming the `TimelockServiceAdapter` `timelockService`, though that's just a matter of taste